tests/gtest/avif_fuzztest_enc_dec_experimental.cc

```diff
@@ -61,14 +61,9 @@ void EncodeDecodeValid(ImagePtr image, EncoderPtr encoder, DecoderPtr decoder) {
   EXPECT_EQ(decoded_image->depth, image->depth);
   EXPECT_EQ(decoded_image->yuvFormat, image->yuvFormat);
 
-  if (decoder->enableParsingGainMapMetadata) {
-    EXPECT_EQ(decoder->gainMapPresent, image->gainMap != nullptr);
-  } else {
-    EXPECT_FALSE(decoder->gainMapPresent);
-  }
-  ASSERT_EQ(decoded_image->gainMap != nullptr, decoder->gainMapPresent);
-  if (decoder->gainMapPresent && decoder->enableDecodingGainMap) {
-    ASSERT_NE(decoded_image->gainMap, nullptr);
+  EXPECT_EQ(decoded_image->gainMap != nullptr, image->gainMap != nullptr);
```

Comment:
To match `decoder->gainMapPresent` in the original code, we should use `decoder->gainMap != nullptr`. Is `decoded_image->gainMap != nullptr` a better check?

This question also applies to the `decoded_image->gainMap != nullptr` check in the next line. The original code also tests `decoder->gainMapPresent` in that if statement.

Reply:
Sorry about the typo. I meant testing `decoder->image->gainMap != nullptr` if we naively convert the `decoder->gainMapPresent` test in the original code.

But I see that the `avifDecoderReadMemory()` call at line 54 ensures that `decoded_image` is a copy of `decoder->image`, so testing `decoded_image->gainMap != nullptr` is equivalent to testing `decoder->image->gainMap != nullptr`.